app/Http/Controllers/CategoryController.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Categories;
+
+class CategoryController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $categories =  Categories::all();
+
+        return view('admin.category.index', compact('categories'));    
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        return view('admin.category.create');
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        $categoryName = $request->name;
+        $categoryContent = $request->content;
+
+        $category = new Categories();
+        $category->name = $categoryName;
+        $category->content = $categoryContent;
+        $category->save();
+
+        return redirect(route('category.index'))->with('status', 'You have successfully added new Category!');
+    }
+
+    /**
+     * Display the specified resource.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function show($id)
+    {
+        $category = Categories::find($id);
+        return view('admin.category.show', compact('category'));
+    }
+
+    /**
+     * Show the form for editing the specified resource.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function edit($id)
+    {
+        $category =  Categories::find($id);
+
+        return view('admin.category.edit', compact('category'));
+    }
+
+    /**
+     * Update the specified resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function update(Request $request, $id)
+    {
+        $categoryName = $request->name;
+        $categoryContent = $request->content;
+
+        $category = Categories::find($id);
+        $category->name = $categoryName;
+        $category->content = $categoryContent;
+        $category->save();
+
+        return redirect(route('category.show', ['id' => $id]))->with('status', 'You have successfully added new Category!');
+    }
+
+    /**
+     * Remove the specified resource from storage.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function destroy($id)
+    {
+        $category = Categories::find($id);
```

Comment:
@jeremiahC Please research about type-hinting the id directly with the model, so you don't have to get the object by `Categories::find($id)`, and just directly type hint to the parameter.